doc/dev/doc/configuration/sources.md

```diff
@@ -0,0 +1,162 @@
+# Configuration Sources
+
+Configuration follows the [Microsoft.Extensions.Configuration](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration)
+patterns. The Hazelcast client configuration is represented by the @Hazelcast.HazelcastOptions class. When simply instantiated, this
+class contains the default options (i.e. it does not even read the options file):
+```csharp
+var options = new HazelcastOptions();
+```
+
+For anything more realistic though, different approaches are available, as detailed below.
+
+This page does not document the options themselves. Options that can be configured are fully documented on the [Options](options.md) page.
+
+### Simple Environment
+
+In a simple, non-hosted environment without dependency injection, options need to be *built* using the
+@Hazelcast.HazelcastOptionsBuilder:
+
+```csharp
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var options = new HazelcastOptionsBuilder.With(args).Build();
+    }
+}
+```
+
+This will determine the application environment (`<env>`) from the `DOTNET_ENVIRONMENT` and `ASPNETCORE_ENVIRONMENT` variables (or, if not specified, default to `Production`), and then gather configuration keys from the following ordered sources:
+
+* `appsettings.json` file
+* `appsettings.<env>.json` file
+* Environment variables (using double-underscore separator, e.g. `hazelcast__clientName`)
+* Command line arguments (using colon separator, e.g. `hazelcast:clientName`)
+* `hazelcast.json` file
+* `hazelcast.<env>.json` file
+* Hazelcast-specific environment variables (using dot separator, e.g. `hazelcast.clientName`)
+* Hazelcast-specific command line arguments (using dot separator, e.g. `hazelcast.clientName`)
+* Optional in-memory key/values
+
+The Hazelcast-specific sources for environment variables and command line arguments only exist to support the non-standard dot separator, and complement the original sources.
+
+The @Hazelcast.HazelcastOptions.Build method supports overriding the name and location of the `hazelcast.json` and
```

Comment:
I assume this @ sign should apply some changes, i.e. create link to API reference or similar? In the server it appears as plaintext: 
![image](https://user-images.githubusercontent.com/4650832/130372447-1ec96a3f-85fc-4dae-8c6c-39988ad58546.png)

Reply:
my mistake, fixing

Reply:
What's the fix here? Will @ now create links to type documentation?

Reply:
The `@` thing creates a direct link to the API documentation - if it can - otherwise it appears as plaintext. Was not working because it was pointing to a non-existent API - fixing